Adds upcoming releases preview section. (#5063)
gravitational/teleport: docs/5.0/preview/upcoming-releases.md

---
title: Teleport Upcoming Releases
description: A timeline of teleport upcoming releases.
---

# Upcoming Teleport Releases

> The teleport team delivers a new major release every 3 months.

Reviewer: Kubernetes has been doing 4 releases per year for years, and now is moving to [3 releases per year](https://github.com/kubernetes/sig-release/issues/1290) since users find it difficult to keep up (we know this from first hand experience). Not saying we should do this now, just something to think about for future.

Author: That's probably because it's so hard to upgrade as you know :) Hopefully our #3121 will let us push releases even sooner.

## Teleport 6.0 "San Diego"

The team open sources role based access controls, implements Database Access, Session Termination
and Access Workflows UI.

### Release Schedule

|Version              | Date              | Description
|---------------------|-------------------|---------------------------
|First alpha          | Jan 15th, 2021    | Good for testing and demos.
|First beta           | Feb 1st, 2021     | Deploy on staging.

> |Release              | March 1st, 2021   | Good to go for production.

Reviewer 2: For Github Milestones, should the "Milestone" date be this one, or the first beta date?

Author: the release one, people usually care more about it.

### Features

You can find the full list of fixes and features in the
[Github milestone](https://github.com/gravitational/teleport/milestone/33).

|Feature                              | Editions          | Description
|-------------------------------------|-------------------|-----------------------------------
| Database Access                     | All               | SSO into PostgreSQL or MySQL. Use AWS Aurora RDS or on-premises. Read [more here](./database-access.md).
| OSS RBAC                            | OSS               | Open source role based access controls. Check out the [design doc](https://github.com/gravitational/teleport/blob/master/rfd/0007-rbac-oss.md) and [issue 4136](https://github.com/gravitational/teleport/issues/4136).

> | Terraform Provider                  | All               | Configure teleport without UI using Terraform provider. More details [here](https://github.com/gravitational/teleport-plugins/projects/3#card-49866475).

Reviewer: Configure **T**eleport

| Dual Authorization Workflows        | Enterprise, Cloud | Request multiple users to review and approve access requests. Find out more in [issue 5007](https://github.com/gravitational/teleport/issues/5007).
| U2F for Kubernetes and SSH sessions | All               | Adds an option to authorize with 2nd factor when connecting to a node/k8s cluster. Details in [issue 3828](https://github.com/gravitational/teleport/issues/3878).
| Access Workflows UI                 | Enterprise, Cloud | Review access requests and assume roles in the UI. Some mockups are in [issue 4937](https://github.com/gravitational/teleport/issues/4937).

> | Client libraries and API            | All               | Use Golang to create access workflows. Review the [design doc](https://github.com/gravitational/teleport/pull/4746) and [issue 4763](https://github.com/gravitational/teleport/issues/4763).

Reviewer 3: ```suggestion
| Client libraries and API            | All               | Use Go to create access workflows. Review the [design doc](https://github.com/gravitational/teleport/pull/4746) and [issue 4763](https://github.com/gravitational/teleport/issues/4763).
```

## Semantic Versioning

Teleport follows [semantic versioning](https://semver.org/) for pre-releases and releases.

**Pre-releases**

> Pre-releases have suffixes `-alpha`, `-beta` and `-rc`.

Reviewer 3: In the RFD, I proposed only using `-dev` and `-rc.X` suffixes: https://github.com/gravitational/teleport/blob/52fdfa90ffba6cb10714718193aa945a95a50d04/rfd/0012-teleport-versioning.md#details
There wouldn't be any `-alpha` or `-beta`.

If we agree on the RFD, this should be updated to match.

Author: Sorry, I missed that part. Let's do `alpha` and `beta` please. We have to communicate differences in expectation.

They are not ready for production:

* You can use alpha releases such as `5.0.1-alpha.1` for trying new features.
  Things can break and new changes may not be backwards-compatible.

* Teleport `beta` releases, such as `5.0.1-beta.2` are suitable for staging environments.

>   We are unlikely to change the APIs while we are ironing bugs and UX glitches.

Reviewer 4: ```suggestion
  We are unlikely to change the APIs while we are ironing out bugs and UX glitches.
```

> * We mark release candidates as `5.0.1-rc.1` when finished coding and bug-fixing.

Reviewer 4: ```suggestion
* We mark release candidates as `5.0.1-rc.1` when coding and bug fixes are finished.
```

>   Team is going through the manual test plan to find any regressions.

Reviewer 4: ```suggestion
  The team is going through the manual test plan to find any regressions.
```
